### PR TITLE
GetAllContents performance increase

### PR DESCRIPTION
**Pull request by ninjanomnom**
It seems a self referential method works better than the list method, even with the proc call costs.

profiling

(In case you dont remember, the order from left to right is: `Self CPU` `Total CPU` `Real Time` `Calls`)

The new GetAllContents had to be wrapped because it calls itself, thus both had to be wrapped to maintain parity in profiling.

### CODE REVIEW

**Comment by ninjanomnom** [comment]
I'll remove the testing code once maintainers check it over

**Comment by Qustinnus** [comment]
what a lad

**Comment by ninjanomnom** [comment]
Why do we have a filter built in to GetAllContents when only 1 thing uses it by the way. Shouldnt the list just be filtered after by the code that needs it using the typecache already there?

**Comment by lzimann** [comment]
@ninjanomnom if it's not used then probably yes.
the new getallcontents looks good

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -503,6 +503,14 @@ Turf and target are separate in case you want to teleport some distance from a t
 	return y
 
 /atom/proc/GetAllContents(list/ignore_typecache)
+	if(prob(50))
+		return OldGetAllContents(ignore_typecache)
+	return NewGetAllContents(ignore_typecache)
+	
+/atom/proc/OldGetAllContents(a)
+	return _OldGetAllContents(a)
```
Comment (Cyberboss) [inline code/__HELPERS/unsorted.dm]: Go ahead and get rid of this now

**Review by Cyberboss — DISMISSED** [review]

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -444,7 +444,7 @@
 /turf/proc/empty(turf_type=/turf/open/space, baseturf_type, list/ignore_typecache, forceop = FALSE)
 	// Remove all atoms except observers, landmarks, docking ports
 	var/static/list/ignored_atoms = typecacheof(list(/mob/dead, /obj/effect/landmark, /obj/docking_port, /atom/movable/lighting_object))
-	var/list/allowed_contents = typecache_filter_list_reverse(GetAllContents(ignore_typecache), ignored_atoms)
+	var/list/allowed_contents = typecache_filter_list_reverse(GetAllContents(), ignored_atoms | ignore_typecache)
```
Comment (ninjanomnom) [inline code/game/turfs/turf.dm]: double check this, I'm 90% sure this is fine

Comment (lzimann) [reply code/game/turfs/turf.dm]: should be fine

**Pull request merged**